src/lib/viewers/image/ImageViewer.js

```diff
@@ -507,6 +508,22 @@ class ImageViewer extends ImageBaseViewer {
         this.annotator.toggleAnnotationMode(nextMode);
         this.processAnnotationModeChange(nextMode);
     }
+
+    handleAnnotationCreateEvent({ annotation: { id } = {}, meta: { status } = {} }) {
+        // Only on success do we exit create annotation mode. If error occurs,
+        // we remain in create mode
+        if (status === 'success') {
+            this.annotator.emit('annotations_active_set', id);
+        }
+    }
+
+    initAnnotations() {
+        super.initAnnotations();
+
+        if (this.areNewAnnotationsEnabled() && this.annotationControls) {
+            this.annotator.addListener('annotations_create', this.handleAnnotationCreateEvent);
```

Comment:
Does this listener need to be removed at some point?

Reply:
I don't think so, `BaseViewer.destroy` takes care of it by calling `this.annotator.removeAllListeners()`